httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/ClientHttp2StreamMultiplexer.java

```diff
@@ -88,9 +88,8 @@ Http2StreamHandler createRemotelyInitiatedStream(
         final HttpCoreContext context = HttpCoreContext.create();
         context.setAttribute(HttpCoreContext.SSL_SESSION, getSSLSession());
         context.setAttribute(HttpCoreContext.CONNECTION_ENDPOINT, getEndpointDetails());
-        if (originalHandler != null && originalHandler instanceof ClientHttp2StreamHandler) {
```

Comment:
@ato What if passed the original request message (or its request URI) instead of linking the contexts?